src/track/globaltrackcache.cpp

```diff
@@ -172,8 +172,13 @@ void GlobalTrackCache::destroyInstance() {
 void GlobalTrackCache::deleter(Track* pTrack) {
     DEBUG_ASSERT(pTrack);
     // Any access to pTrack is forbidden!! Due to race condition
-    // this pointer might already have been deleted! Only the
-    // cache knows this.
+    // this pointer might already have been deleted! This happens
+    // when a previous invocation is outpaced by a following
+    // invocation that will already delete the object behind the
+    // pointer. Tracks might be revived before being deleted and
```

Comment:
This sound like we doing something wrong. Normally smart pointer weak pointer pair is designed to prevent exactly this.
Do we keep a plain pointer in some places which destroys this nature? 
The reference counter of the calling smart pointer should be already zero and the weak pointer can't return a valid pointer in this stage, right? How can the global atomic reference counter ever increment from 0?